### ISSUE TITLE
[TW-888] Unblocked tasks do not show up on cue

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#923]
_Sander Marechal on 2011-01-12T02:51:58Z says:_ 

 When a task is blocked by another task, and the blocker is finished, the now unblocked task does not show up on the next invocation of task. It only starts showing on the task invocation after that. It appears that it requires two garbage collection runs for a newly unblocked task to show up. It should only require one GC run, or preferably none at all. Example:

$ task add foo
Created task 1.

$ task add bar depends:1
Created task 2.

$ task
(shows task 1)

$ task 1 done
Completed 1 'foo'.

$ task
(shows no tasks)

$ task rc.gc:off
(still shows no tasks, gc is not run)

$ task
(shows task 2)

It should require at most one garbage collection run for task 2 to show up.

Preferably, it should not require garbage collection at all. Tasks should be unblocked at the moment parent tasks are finished or removed, not when old tasks are pruned and the rest of the tasks is renumbered. I don't know if that is possible, but it would be really nice to have.

PS: I'm using Taskwarrior 1.9.4 from Git.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2011-01-12T02:51:58Z
Modified: 2014-02-09T02:11:50Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-01-12T07:38:50Z says:_ 

 Thank you Sander.

**Comment by taskwarrior** [comment]
_Cory Donnelly on 2011-01-22T12:49:47Z says:_ 

 I couldn't recreate this issue, has it been addressed since?

**Comment by taskwarrior** [comment]
_Sander Marechal on 2011-01-23T04:22:45Z says:_ 

 I did a git pull and tried to build the latest version to check myself, but Taskwarrior seems to have switched to CMake. How do I build it with CMake?

**Comment by taskwarrior** [comment]
_Federico Hernandez on 2011-01-23T07:57:43Z says:_ 

 To build task with cmake follow the instructions in the INSTALL file.

Backup your data, 1.9.4 is still under development. So functionality might be broken or features/bug fixes are not fully implemented yet.

And please report any problems you discover to our support email address including the sha1 from your build. You see it with 'task diag'. The bug tracker should remain for released versions concerning bugs. Thx.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-02-12T01:07:54Z says:_ 

 I think this was a transient result from breakage with the 'all' report, which has since been fixed.  I cannot reproduce.
